### PR TITLE
Add workaround for pkcs11

### PR DESCRIPTION
**Pull request by Radioo**
This is a workaround for an issue that happens when you include the pkcs11 module. The compiler will complain about three missing headers even in amalgamation builds.

### CODE REVIEW

**Comment by Tectu** [comment]
Interesting... Do you have any idea as to why this issue occurs? This smells like something that would potentially require an upstream fix instead (i.e. a fix in Botan itself).
I'll be looking into this. Could you tell me which version(s) of Botan you encountered this issue with?

**Comment by Radioo** [comment]
https://github.com/randombit/botan/issues/1447
https://github.com/randombit/botan/issues/976

There have been discussions about this in the past, but it seems nothing has been done about the issue since. I'd give it a shot of fixing it but I'm afraid I'm not qualified to tackle that with my current skill.
I was building the most recent version at the time which was `3.1.1`.
My cmake generation section looks like this:
```
find_package(
        Botan 3.1.1
        REQUIRED
)

botan_generate(
        botan_pkcs11
            pkcs11
)
```
Building on Windows with MSVC.

**Comment by Tectu** [comment]
Thank you for clarifying this!
This certainly seems like an interesting solution. Personally, I'd agree that this could be considered a "bug" in Botan. However, given the purpose/goal/mission of this CMake integration I also think that it's pretty reasonable to include your fix.
I will sleep one more night over it and most likely merge it.

File: cmake/FindBotan.cmake [hunk 1]
```diff
@@ -23,10 +23,10 @@ include(FetchContent)
 
 # Find python
 find_package(
-    Python
-    COMPONENTS
+        Python
+        COMPONENTS
         Interpreter
-    REQUIRED
+        REQUIRED
```
Comment (Tectu) [inline cmake/FindBotan.cmake]: Can you please remove these whitespace changes?

File: cmake/FindBotan.cmake [hunk 2]
```diff
@@ -37,18 +37,18 @@ set(DOWNLOAD_URL https://github.com/randombit/botan/archive/refs/tags/${Botan_VE
 
 # Just do a dummy download to see whether we can download the tarball
 file(
-    DOWNLOAD
-    ${DOWNLOAD_URL}
-    STATUS download_status
+        DOWNLOAD
+        ${DOWNLOAD_URL}
+        STATUS download_status
```
Comment (Tectu) [inline cmake/FindBotan.cmake]: Can you please remove these whitespace changes?

File: cmake/FindBotan.cmake [hunk 3]
```diff
@@ -37,18 +37,18 @@ set(DOWNLOAD_URL https://github.com/randombit/botan/archive/refs/tags/${Botan_VE
...
 if (NOT download_status EQUAL 0)
     message(FATAL_ERROR "Could not download Botan tarball (status = ${download_status}): ${DOWNLOAD_URL}")
 endif()
 
 # Download the tarball
 FetchContent_Declare(
-    botan_upstream
-    URL ${DOWNLOAD_URL}
+        botan_upstream
+        URL ${DOWNLOAD_URL}
```
Comment (Tectu) [inline cmake/FindBotan.cmake]: Can you please remove these whitespace changes?

File: cmake/FindBotan.cmake [hunk 4]
```diff
@@ -74,9 +80,9 @@ function(botan_generate TARGET_NAME MODULES)
 
     # Run the configure.py script
     add_custom_command(
-        OUTPUT botan_all.cpp botan_all.h
-        COMMENT "Generating Botan amalgamation files botan_all.cpp and botan_all.h"
-        COMMAND ${Python_EXECUTABLE}
+            OUTPUT botan_all.cpp botan_all.h
+            COMMENT "Generating Botan amalgamation files botan_all.cpp and botan_all.h"
+            COMMAND ${Python_EXECUTABLE}
```
Comment (Tectu) [inline cmake/FindBotan.cmake]: Can you please remove these whitespace changes?

File: cmake/FindBotan.cmake [hunk 5]
```diff
@@ -92,20 +98,33 @@ function(botan_generate TARGET_NAME MODULES)
     set(TARGET ${TARGET_NAME})
     add_library(${TARGET} STATIC)
     target_sources(
-        ${TARGET}
-        PUBLIC
+            ${TARGET}
+            PUBLIC
             ${CMAKE_CURRENT_BINARY_DIR}/botan_all.h
-        PRIVATE
+            PRIVATE
```
Comment (Tectu) [inline cmake/FindBotan.cmake]: Can you please remove these whitespace changes?

File: cmake/FindBotan.cmake [hunk 6]
```diff
@@ -92,20 +98,33 @@ function(botan_generate TARGET_NAME MODULES)
...
-        INTERFACE
+            ${TARGET}
+            INTERFACE
             ${CMAKE_CURRENT_BINARY_DIR}
     )
     set_target_properties(
-        ${TARGET}
-        PROPERTIES
+            ${TARGET}
+            PROPERTIES
```
Comment (Tectu) [inline cmake/FindBotan.cmake]: Can you please remove these whitespace changes?

**Review by Tectu — CHANGES_REQUESTED** [review]

**Comment by Radioo** [comment]
Sorry about that, no idea how those whitespaces got changed.

**Review by Tectu — APPROVED** [review]

**Comment by Tectu** [comment]
Thank you for your contribution!

**Pull request merged**